### PR TITLE
[MM-49224] Change at mention button implementation back to anchors 

### PR DESCRIPTION
**Pull request by cyrilzhang-mm**
#### Summary

I originally changed at mention buttons to use `<button>` instead of `<a>` because it easily handled both enter and space key press, but it causes wrapping issues if the mention is very long. This PR changes them back to `<a>` with `role='button'` and adds the proper enter and space key handling. 

#### Ticket Link

[Jira ticket](https://mattermost.atlassian.net/browse/MM-49224)

#### Screenshots

|  before  |  after  |
|----|----|
| Screen Shot 2022-12-19 at 12 32 41 PM  |  Screen Shot 2022-12-19 at 12 40 17 PM   |

#### Release Note

```release-note
Fixed bug where long group mentions and user mentions don't wrap properly. 
```

### CODE REVIEW

File: components/user_group_popover/group_member_list/group_member_list.tsx [hunk 1]
```diff
@@ -170,7 +170,10 @@ const GroupMemberList = (props: Props) => {
                     key={user.id}
                     role='listitem'
                 >
-                    <UserButton onClick={() => showUserOverlay(user)}>
+                    <UserButton
+                        onClick={() => showUserOverlay(user)}
+                        aria-haspopup={true}
```
Comment (BenCookie95) [inline components/user_group_popover/group_member_list/group_member_list.tsx:175]: Should this be `dialog` since it opens a popover?

**Review by BenCookie95 — CHANGES_REQUESTED** [review]

File: components/user_group_popover/group_member_list/group_member_list.tsx (hunk 1 above)
Comment (cyrilzhang-mm) [reply components/user_group_popover/group_member_list/group_member_list.tsx]: That's right. Thanks for catching that.

**Review by BenCookie95 — APPROVED** [review]
LGTM!

**Review by mkraft — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working fine

**Pull request merged**